apis/kubedb/v1/types.go

```diff
@@ -190,7 +190,11 @@ type SystemUserSecretsSpec struct {
 
 type SecretReference struct {
 	core.LocalObjectReference `json:",inline,omitempty"`
-	ExternallyManaged         bool `json:"externallyManaged,omitempty"`
+	// Recommendation engine will generate RotateAuth opsReq using this field
+	RotateAfter metav1.Duration `json:"rotateAfter,omitempty"`
```

Comment:
We should use pointer type here, it would be easier to check nil if user not provided -
ref: https://github.com/kubeops/supervisor/blob/eaebc7625a22c06e54399b3b29a7bea8ef5d7bc8/apis/supervisor/v1alpha1/recommendation_types.go#L59